### DISCUSSION TITLE
TextBox with "VariableSelectionHighlight"

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#11719] [Q&A]
I need a control where I can mark specific ranges in a text as a variable. I would love to specific e.g. "character 5-10 please be red" and I need to be able to register if a "variable" has been clicked.

I tried my own implementation using Skia, but it has many drawbacks. Basically I have to mimic a TextBox from scratch and render the text and the variables on my own. Thie brings some problems. The main one is: OnKeyDown does give me the pressed key - but it could be anything. A control key, or a real key. It is quite hard to get the unicode representation of a pressed key without PInvoking into Win32.dll.

So I am thinking of other ways and tried to understand the TextBox-Implementation. I have some questions:
- where does the implementation gets the "printable character" from a keypress? Is that a function I could call, too?
- is there a way to get the coordinates of a range of text to draw something at that location on my own?
- Is there any other control I could use to support "variables" or "buttons/objects" within a text? RichEditBox is not supported, yet, so this does not work unfortunately.

Thanks for any hint or idea!

**Comment by jeromelaban** [comment]
At this time, such an implementation would need to be native, as Uno does not provide rich editable content. With regards to specifics of your question, try creating an implementation using WinAppSDK and we may determine if there APIs that we can implement instead.

**Reply by TopperDEL** [reply]
Thanks! I found a way using SkiaSharp and [RichTextKit](https://github.com/toptensoftware/RichTextKit).
